src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -240,12 +249,14 @@ - (void)writeMessage:(NSData *)message withErrorHandler:(void (^)())errorHandler
     // Resume the request writer.
     GRPCCall *strongSelf = weakSelf;
     if (strongSelf) {
-      strongSelf->_requestWriter.state = GRXWriterStateStarted;
+      @synchronized(_requestWriter) {
```

Comment:
As described above, all access to the _requestWrite's state, both getting and setting, have to be _synchronized on the same object.

@synchronized(strongSelf) { } in this case to match the @synchronized(self) above (they are the same object, its just that assigning to strongSelf can set strongSelf to nil if self got deallocated before the callback was called.)